cli/src/commands/parse.rs

```diff
@@ -61,11 +43,11 @@ pub fn try_get_packages(path: &Path) -> Result<(Vec<PackageDescriptor>, PackageT
 
     let data = read_to_string(path)?;
 
-    for (name, parser) in LOCKFILE_PARSERS.iter() {
-        if let Ok(pkgs) = parser.parse(data.as_str()) {
+    for format in LockfileFormat::iter() {
+        if let Ok(pkgs) = format.parser().parse(data.as_str()) {
             if !pkgs.is_empty() {
-                print_user_success!("Identified lockfile type: {}", name);
-                return Ok((pkgs, parser.package_type()));
+                print_user_success!("Identified lockfile type: {}", format);
+                return Ok((pkgs, format.parser().package_type()));
             }
```

Comment:
The `parser()` function is called twice here. I think it would be cleaner to extract that into a `parser` variable and reuse it.